### DISCUSSION TITLE
Windows Server hosted runner can't handle long make command.

### DISCUSSION THREAD

**Discussion by wallybarnum** [community/community#123346] [Actions]
### Select Topic Area

Question

### Body

I am building a Make project with C and C++ files using Make for Windows as the make.exe and a gcc for arm toolchain.
When it comes time to link, the resultant command to gcc and the linker is a little over 12k chars long due to the number of object files and their paths.

If I build it on a Windows 11 desktop and powershell calling Make for windows, it builds fine.
If I build this on an Ubuntu hosted runner and its built in make with my gcc toolchain, it builds fine.
**If I build it on a Windows Server-latest runner in git hub actions, specifying powershell and calling Make for Windows, it doesn't build and its clear that the command is being truncated.** 

Is there a way to configure the windows server host in GitHub actions to have its powershell accept larger input strings?

Thanks.

**Comment by wallybarnum** [comment]
reopened

**Reply by adewie** [reply]
This might be helpful https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

**Reply by wallybarnum** [reply]
good idea.  just tried that, but unfortunately did not work. I suspect this fixes long file path limitations, but not long powershell command parsing.

**Reply by adewie** [reply]
There are a couple of ways to run cmd commands programmatically that might work. Like doing this in c# or a python equivalent. System.Diagnostics.Process.Start("CMD.exe",strCmdText);
